processing/src/main/java/io/druid/query/timeboundary/TimeBoundaryQuery.java

```diff
@@ -70,21 +73,25 @@ public TimeBoundaryQuery(
         context
     );
 
+    this.dimFilter = dimFilter;
     this.bound = bound == null ? "" : bound;
   }
 
   @Override
-  public boolean hasFilters()
-  {
-    return false;
-  }
+  public boolean hasFilters() { return dimFilter != null; }
```

Comment:
Formatting is off here. You can use the IntelliJ/Eclipse style bundles in https://github.com/druid-io/druid/blob/master/CONTRIBUTING.md to format the code automatically.